build-logic/packaging/src/main/kotlin/gradlebuild/instrumentation/tasks/InstrumentedMetadataMergeTask.kt

```diff
@@ -47,25 +52,40 @@ abstract class FindInstrumentedSuperTypesTask : DefaultTask() {
     @get:OutputFile
     abstract val instrumentedSuperTypesHash: RegularFileProperty
 
+    @get:OutputFile
+    abstract val upgradedProperties: RegularFileProperty
+
+    @get:OutputFile
+    abstract val upgradedPropertiesHash: RegularFileProperty
```

Comment:
:x: Let's add some Javadoc about the purposes of these two output files. Specifically, I'd be interested why there is a need for a separate file for a hash. Might also consider adding similar Javadoc to the other properties. Adding the Javadoc on the task type would also work.

IIUC, the purpose of the `hash` files is to explicitly invalidate custom caching, but it's unclear which cache we are invalidating. Also, this workaround will not be required as soon as we move this work to artifact transforms instead, right? If so, we should spell that out here, too, maybe as a `TODO`.

Reply:
Done.

> Also, this workaround will not be required as soon as we move this work to artifact transforms instead, right?
If we will use just file hashing, then we won't need it. But then we need to make sure json has always stable order of properties.

Reply:
If we use Jackson instead of Gson, it might be easier to make order stable. And since we generate these files in build logic, we can use Jackson.